taskcluster/app_services_taskgraph/transforms/beetmover.py

```diff
@@ -2,74 +2,183 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+import posixpath
 
 from taskgraph.transforms.base import TransformSequence
 from taskgraph.util.schema import resolve_keyed_by
 
-from . import (
-    publications_to_artifact_paths, publications_to_artifact_map_paths
-)
+from . import publications_to_artifact_paths, publications_to_artifact_map_paths
 from ..build_config import get_version
-from ..beetmover import get_maven_bucket
 
 transforms = TransformSequence()
 
 
 @transforms.add
-def build_upstream_artifacts(config, tasks):
+def resolve_keys(config, tasks):
     for task in tasks:
-        module_info = task["attributes"]["buildconfig"]
-        version = get_version(config.params)
-        publications = module_info["publications"]
+        for key in ("worker.action",):
+            resolve_keyed_by(
+                task, key, item_name=task["name"], level=config.params["level"]
+            )
+        yield task
+
+
+def _build_upstream_artifacts(config, task):
+    upstream_artifacts = []
+
+    for label, dep in config.kind_dependencies_tasks.items():
+        if label not in task["dependencies"].values():
+            continue
+
+        paths = sorted(
+            artifact["name"] for artifact in dep.attributes.get("release-artifacts", [])
+        )
+
+        if paths:
+            upstream_artifacts.extend(
+                [
+                    {
+                        "taskId": {"task-reference": f"<{dep.kind}>"},
+                        "taskType": "repackage",
```

Comment:
This seems like one of those features that I wished I knew about before.  Is there documentation somewhere?

Reply:
I sort-of understand `task-reference`, I was wondering about `taskType: repackage`.  Is that specifically for tasks that input artifacts from other tasks and output a repackaged version of them?

Reply:
Yeah, the documentation is still WIP. As for `release-artifacts`, I'm going to upstream this to [mozilla-taskgraph](https://github.com/mozilla-releng/mozilla-taskgraph) at some point and I'll add docs there when I do.

Reply:
Right, upstream-artifacts is something scriptworker / chain-of-trust use to download and validate artifacts from dependencies.